source-container-build/app/test_source_build.py

```diff
@@ -1271,15 +1271,14 @@ def test_source_image_is_resolved(self, run: MagicMock):
 @pytest.mark.parametrize(
     "image_pullspec,expected",
     [
-        ["ubuntu", "ubuntu"],
-        ["reg:3000", "reg:3000"],
-        ["reg:3000/img", "reg:3000/img"],
-        ["reg:3000/img:9.3", "reg:3000/img:9.3"],
-        ["reg:3000/img:9.3@sha256:123", "reg:3000/img@sha256:123"],
-        ["reg/org/img:9.3@sha256:123", "reg/org/img@sha256:123"],
-        ["reg/org/path/img:9.3", "reg/org/path/img:9.3"],
-        ["reg/org/path/img:9.3@sha256:123", "reg/org/path/img@sha256:123"],
+        ["ubuntu", ("ubuntu", "", "")],
+        ["reg:3000", ("reg", "3000", "")],
```

Comment:
Shouldn't the expected result here be `("reg:3000", "", "")`

Reply:
This is an interesting case actually. Both parts `reg` and `3000` are valid for name and tag, even if `.` is included. They are usually recognized as that by tools as well, like skopeo and podman. And the first part is recognized as registry only when it includes `.` and `/` is included in the name. Anyway, this is just a rough parser and not to be a generic parser for image name.

Reply:
ok